noxfiles/ci_nox.py

```diff
@@ -0,0 +1,202 @@
+"""Contains the nox sessions used during CI checks."""
+import nox
+from constants_nox import (
+    CI_ARGS,
+    COMPOSE_FILE,
+    IMAGE_NAME,
+    INTEGRATION_COMPOSE_FILE,
+    RUN,
+    RUN_NO_DEPS,
+    START_APP,
+    WITH_TEST_CONFIG,
+)
+from docker_nox import build_local_prod
+from utils_nox import teardown
+
+RUN_STATIC_ANALYSIS = (*RUN_NO_DEPS, "nox", "-s")
+
+
+@nox.session()
+def check_all(session: nox.Session) -> None:
+    """Runs all of the CI checks."""
+    teardown(session)
+    build_local_prod(session)
+    black(session)
+    isort(session)
+    xenon(session)
+    mypy(session)
+    pylint(session)
+    check_install(session)
+    fidesctl(session)
+    fidesctl_db_scan(session)
+    pytest_unit(session)
+    pytest_integration(session)
+
+
+# Static Checks
+@nox.session()
+def black(session: nox.Session) -> None:
+    """Run the 'black' style linter."""
+    if session.posargs == ["docker"]:
+        run_command = (*RUN_STATIC_ANALYSIS, "black")
+    else:
+        run_command = ("black", "--check", "src", "tests", "noxfiles")
```

Comment:
I know you are running this with `--check` for CI and that makes sense. It would be nice to also have the option to have it run the fix. When running outside of CI if this fails you will immediately run black to fix things so it would cut out a step. Maybe something like having a `black_check_only` and `black` option could work?

Reply:
good point, i thought about this too. Will add options for `black` and `isort` to not run in check mode